.github/DEVELOPMENT.md

```diff
@@ -210,5 +210,37 @@ The `pnpm run test:migrate` script is run in CI to ensure that templating change
 See `.github/workflows/test-migrate.yml`.
 
 > Tip: if the migration test is failing in CI and you don't see any errors, try [downloading the full logs](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/using-workflow-run-logs#downloading-logs).
-> There'll likely be a list of changed files under a message like _`Oh no! Running the migrate script modified some files:`_.
-> You can also try running the test script locally.
+
+##### Migration Snapshot Failures
+
+The migration test uses the [Vitest file snapshot](https://vitest.dev/guide/snapshot#file-snapshots) in `script/__snapshots__/migrate-test-e2e.js.snap` to store expected differences to this repository after running the migration script.
+The end-to-end migration test will fail any changes that don't keep the same differences in that snapshot.
+
+You can update the snapshot file by:
+
+1. Committing any changes to your local repository
+2. Running `pnpm i` and `pnpm build` if any updates have been made to the `package.json` or `src/` files, respectively
+3. Running `pnpm run test:migrate -u` to update the snapshot
+
+At this point there will be some files changed:
+
+- `script/__snapshots__/migrate-test-e2e.js.snap` will have updates if any files mismatched templates
+- The actual updated files on disk will be there too
+
+If the snapshot file changes are what you expected, then you can commit them.
+The rest of the file changes can be reverted.
+
+> [🚀 Feature: Add a way to apply known file changes after migration #1184](https://github.com/JoshuaKGoldberg/create-typescript-app/issues/1184) tracks turning the test snapshot into a feature.
+
+##### Unexpected File Modifications
```

Comment:
maybe we can bring this to the top! So they know about the error first and then how to update it?

Reply:
I like the idea of moving the info closer to where users will read it! I'm not sure what you mean though - are you saying to move this section to the start of the `## Setup Scripts` area? ...the top of the `### The Migration Script` area? `#### Testing the Migration Script`?

Reply:
This I meant: `#### Testing the Migration Script`

Let me know if it makes sense.

Reply:
Cool yeah it does, thanks!

My holdup was that I'm hesitant to put direct _solutions_ to questions before _answering_ the question in the first place. Like, the docs can explain how to resolve a snapshot difference, but if someone doesn't know what the snapshots are or why they're failing then it'll be harder for them to figure out how to do that change right.

I think the compromise in e5e41942313ae9a4cb71b000abc5dbc5368a8018 solves for this: it adds a specific _"seeing this? solution is there!"_ link to `#### Testing the Migration Script`. I think this way folks will know there's a solution, and also that it bypasses some of the docs.